### PR TITLE
PYTHON-4903 Adds typing overloading to bson.binary.Binary.from_vector

### PR DESCRIPTION
**Pull request by caseyclements**
Follows suggestion from @ShaneHarvey that adding typing oveloading would make clear that dtype  is a required argument when the first argument, vector, is a list, not at BinaryVector instance.

### CODE REVIEW

**Review by blink1073 — CHANGES_REQUESTED** [review]
We should also remove the **BETA** and add a changelog entry

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Do we have a test that `from_vector(BinaryVector(), dtype=...)` raises an error and requires a type-ignore?

File: test/test_bson.py [hunk 1]
```diff
@@ -802,6 +802,13 @@ def test_vector(self):
         assert float_binary == Binary.from_vector(
             BinaryVector(list_vector, BinaryVectorDtype.FLOAT32)
         )
+        # Confirm kwargs cannot be passed when BinaryVector is provided
+        self.assertRaises(
```
Comment (ShaneHarvey) [inline test/test_bson.py:806]: Can you write this as a `with self.assertRaises():` so we can confirm a "type-ignore" is detected?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/test_bson.py (hunk 1 above)
Comment (caseyclements) [reply test/test_bson.py:806]: Sure can. How do we confirm that a type-ignore is detected?

```
        with self.assertRaises(ValueError):
            Binary.from_vector(
                BinaryVector(list_vector, BinaryVectorDtype.PACKED_BIT, padding),
                dtype=BinaryVectorDtype.PACKED_BIT
            )
```

Comment (caseyclements) [reply test/test_bson.py:806]: Oh.... I see. This will fail linting without it.

Comment (ShaneHarvey) [reply test/test_bson.py]: Yep that's what I was thinking!

**Review by ShaneHarvey — APPROVED** [review]

File: test/test_bson.py (hunk 1 above)
Comment (caseyclements) [reply test/test_bson.py]: TIL. Thanks @ShaneHarvey !  This was a very cool lesson.

**Review by blink1073 — APPROVED** [review]

**Pull request merged**